src/tactic/rw_hint/default.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import tactic.rewrite_search.discovery
+import tactic.rewrite_all
+
+open tactic.rewrite_all.congr
+open tactic.rewrite_search.discovery
+
+namespace tactic
+
+/-- Generate a list of `rw ...` strings, which represent successful rewrites of the current goal. -/
+-- TODO perhaps run `solve_by_elim` on resulting goals,
+-- and put any rewrites where this succeeds at the top of the list?
+-- TODO some suggestions will appear for any equality goal, but are stupid:
+--   Try this: rw eq_comm
+--   Try this: rw ←option.some_inj
+--   Try this: rw ←eq_comm
+--   Try this: rw ←eq_iff_eq_cancel_right
+--   Try this: rw ←eq_iff_eq_cancel_left
+--   Try this: rw ←option.some.inj_eq
+--   Try this: rw ←heq_iff_eq
+-- Is there a sensible way to discard these? Otherwise we could blacklist them?
```

Comment:
At the least, you should probably filter internal declarations, i.e. those with a name component starting with `_`.

Reply:
Done.